### ISSUE TITLE
Node.js executable crashes with an access violation when a smart card reader is inserted or removed on Windows.

### ISSUE THREAD

**Issue by Secalot** [node-usb/node-usb#285]
#### Symptom:
Windows 10 x64, Node 9.5.0.
Run any script that includes the node-usb module, for example

``` js
import usb from 'usb'

const test = async () => {}
setInterval(() => {
  console.log('!')
}, 1000)
```
Insert or remove a USB smart card reader while the script is running. Node executable will crash with an access violation.

#### Rootcause:
The libusb version used by node-usb has a bug in device addition/removal detection code under Windows.
The access violation happens inside the parse_device_interface_path function in the ".\libusb\libusb\os\windows_usb.c" file if the interface_path parameter passed to it starts with a "{" symbol. With that the "len" variable is -1,  but as "len" is declared as an unsigned, the check for it to be less than zero is aways false and the loop at the end of the function executes until 0xFFFFFFFF, causing an access violation.
As smart cards have interface paths like "{892EDE5E-BE49-443c-A0B3-005D74F2D69C}#ScFilter#9&3a0c6d5&0&01#{d86354cc-a2ac-4223-95b9-2e48ce154434}", the error is hit.

**Comment by mildsunrise** [comment]
I think this can be closed in favor of #386

**Comment by thegecko** [comment]
Thanks, please re-open if required
